File: app/src/main/java/jp/gr/java_conf/foobar/testmaker/service/models/StructQuestion.java
package jp.gr.java_conf.foobar.testmaker.service.models;

import java.io.Serializable;

import jp.gr.java_conf.foobar.testmaker.service.Constants;

/**
 * Created by keita on 2016/06/02.
 */
public class StructQuestion implements Serializable {

    public int type;
    public String answer;
    public String question;
    public String[] others;
    public String[] answers;
    public String imagePath;
    public String explanation;
    public boolean light;
    public boolean auto;

    public StructQuestion(String q, String a) {
        question = q;
        answer = a;
        explanation = "";
        light = false;
        others = new String[3];
        answers = new String[0];
        type = Constants.WRITE;
        auto = false;
    }

    public StructQuestion(String q, String a[]) {
        question = q;
        answer = "";

        for (String answer: a
             ) {
            this.answer += answer + " ";
        }

        explanation = "";

        light = false;
        answers = a;
        others = new String[0];
        type = Constants.COMPLETE;
        auto = false;
    }

    public StructQuestion(String q, String a, String[] o) {
        question = q;
        answer = a;
        others = o;
        answers = new String[0];
        type = Constants.SELECT;
        explanation = "";
        light = false;
        auto = false;

    }

    public void setExplanation(String explanation) {
        this.explanation = explanation;
    }

    public void setImagePath(String path) {
        this.imagePath = path;
    }

    public void setAuto(boolean auto) {
        this.auto = auto;
    }
}

File: app/src/main/java/jp/gr/java_conf/foobar/testmaker/service/models/StructQuestion.kt
package jp.gr.java_conf.foobar.testmaker.service.models

import java.io.Serializable

import jp.gr.java_conf.foobar.testmaker.service.Constants

/**
 * Created by keita on 2016/06/02.
 */
class StructQuestion : Serializable {

    var type: Int = 0
    var answer: String
    var question: String
    var others: Array<String>
    var answers: Array<String>
    var imagePath: String
    var explanation: String
    var light: Boolean = false
    var auto: Boolean = false
    private var order: Int = 0
    var isCheckOrder = false

    constructor(q: String, a: String) {
        question = q
        answer = a
        imagePath = ""
        explanation = ""
        light = false
        others = arrayOf("")
        answers = arrayOf("")
        type = Constants.WRITE
        auto = false
    }

    constructor(q: String, a: Array<String>) {
        question = q
        answer = ""

        for (answer in a) {
            this.answer += "$answer "
        }

        imagePath = ""
        explanation = ""
        light = false
        answers = a
        others = arrayOf("")
        type = Constants.COMPLETE
        auto = false
    }

    constructor(problem: String, answers: Array<String>, others: Array<String>) {
        question = problem
        answer = ""
        for (answer in answers) {
            this.answer += "$answer "
        }
        this.others = others
        this.answers = answers
        type = Constants.SELECT_COMPLETE
        imagePath = ""
        explanation = ""
        light = false
        auto = false
    }

    constructor(q: String, a: String, o: Array<String>) {
        question = q
        answer = a
        others = o
        answers = arrayOf("")
        type = Constants.SELECT
        imagePath = ""
        explanation = ""
        light = false
        auto = false

    }

    fun setOrder(order: Int) {
        this.order = order
    }
}
